frontend/src/store/controller/reducers.ts

```diff
@@ -46,12 +51,11 @@ const messageReducer = <T extends PBMessage>(
   switch (action.type) {
     case STATE_UPDATED:
       if (action.messageType === messageType) {
+        if (messageType === MessageType.NextLogEvents) {
+          return mergeEventList(action.state as NextLogEvents, state as NextLogEvents) as T;
+        }
         if (action.messageType === MessageType.RotaryEncoder) {
-          const updatedState = calculateStepDiff(
-            state as RotaryEncoder,
-            action.state as RotaryEncoder,
-          );
-          return updatedState as T;
+          return calculateStepDiff(action.state as RotaryEncoder, state as RotaryEncoder) as T;
```

Comment:
Is there any reason you've put these `MessageType` case handlers inside the general `messageReducer`, rather than making their own reducers? It's not really consistent with the spirit of how `messageReducer` is parameterized - but I'm not sure if there is a way to compose two reducers (e.g. feed the output of `messageReducer<NextLogEvents>(MessageType.NextLogEvents, NextLogEvents) into a reducer which calls `mergeEventList`), either. If we want to fix this, we can do it in a future PR - this is okay for now.

Reply:
Has been fixed on #190